File: frontend/src/components/types/tea.ts
import { Shop } from "@/components/types/shop.ts";
import { TeaIngredient } from "@/components/types/teaIngredient.ts";

export interface Tea {
  id: number;
  name: string;
  description?: string;
  created_at: string;
  shop: {
    id: Shop["id"];
    name: Shop["name"];
  };
  teaIngredients: TeaIngredient[];
}

export interface TeaWithQualities extends Tea {
  qualities: {
    id: number;
    teaQuality: {
      name: string;
      id: number;
    };
    value: number | string;
  }[];
}

File: backend/herbatka/models/tea.py
from django.db import models
from .shop import Shop
from .tea_ingredient import TeaIngredient


class Tea(models.Model):
    name = models.CharField(max_length=255)
    description = models.TextField(blank=True)
    created_at = models.DateTimeField(auto_now_add=True)

    shop = models.ForeignKey(Shop, on_delete=models.CASCADE, related_name="teas")
    tea_ingredients = models.ManyToManyField(TeaIngredient, related_name="teas", blank=True)

    def _str_(self):
        return self.name
